src/main/java/seedu/multitasky/model/entry/EntryList.java

```diff
@@ -148,6 +149,30 @@ public void updateEntry(ReadOnlyEntry target, ReadOnlyEntry editedEntry)
 
     // @@author A0126623L
     /**
+     * Checks if duplicate entries are present after editing an entry. The entry to be edited
+     * is excluded from the duplicates-check.
+     * @param target        entry to edit
+     * @param editedEntry
+     * @return boolean
+     */
+    private boolean duplicatesPresentAfterEditing(ReadOnlyEntry target, ReadOnlyEntry editedEntry) {
+        ArrayList<Entry> prospectiveInternalList = createInternalListCopyWithoutTarget(target);
+        return prospectiveInternalList.contains(editedEntry);
+    }
+
+    // @@author A0126623L
+    private ArrayList<Entry> createInternalListCopyWithoutTarget(ReadOnlyEntry target) {
+        ArrayList<Entry> copiedInternalList = new ArrayList<Entry>();
+        for (Entry e : internalList) {
+            if (!(e == target)) {
```

Comment:
I'll stick with the `==` operator to make explicit to readers of the code that I'm comparing reference here.